fem/restriction.hpp

```diff
@@ -21,20 +21,10 @@ namespace mfem
 class FiniteElementSpace;
 enum class ElementDofOrdering;
 
-/// Abstract base class that defines an interface for element restrictions.
-class ElementRestrictionOperator : public Operator
-{
-public:
-   /// @brief Add the E-vector degrees of freedom @a x to the L-vector degrees
-   /// of freedom @a y.
-   void AddMultTranspose(const Vector &x, Vector &y,
-                         const double a = 1.0) const override = 0;
-};
```

Comment:
Was there no need for this class after all, removing this class seems a bit unrelated to the purpose of this PR.

Reply:
The `ElementRestrictionOperator` was needed for `AddMultTranspose` before PR #3075, which has made this abstract base class obsolete (the interface is now in `Operator` itself).

I don't see the harm in leaving it in this PR

Reply:
I am not against having this in this PR (though having atomic PR is always better), but can you remind me why it was introduced, and why you're judging now that it's unneeded?

Reply:
Logically, it is good to have a common base class for all element-restriction classes, even if that does not do anything specific at the moment (in the future we may want to add methods, e.g. to query the E-vector layout when using mixed meshes or variable orders). Also, removing the base class changes the API in a way that is not backward compatible -- this needs to be documented in the CHANGELOG if you decide to keep this change.

Reply:
How about we decouple this change from the rest of the PR since it's orthogonal. We can have a separate PR or discussion about this proposed change.

Reply:
Agreed, done in d2e2b5d. See #3599 for the separate PR.